### DISCUSSION TITLE
See size of node in Inspector

### DISCUSSION THREAD

**Discussion by CubeOfCheese** [godotengine/godot-proposals#9681] [Editor]
I'm surprised that I can't find the size of a node in the Inspector. I am trying to create multiple collision shapes of the same width, but there's no way for me to see their width in the editor. I know I could run the project and `print(node.size)` but 1. I shouldn't have to and 2. running the project each time I make a tweak just to see how much I tweaked it by is really annoying. Why can't I see and set a width and height property in the Inspector like I can with position?

I'm also aware of the ruler tool, but that requires me to click the sides of the node with accuracy and doesn't allow me to set the width by typing a number like I can with position.

Is there some kind of technical limitation that prevents this or am I missing something? This seems like an obvious and important feature.

**Comment by AThousandShips** [comment]
Only `Control` nodes have a `size`, so I'm unsure what you're using collision shapes for, other nodes don't have a size, what node are you calling `print(node.size)` on?

**Reply by Squidgical** [reply]
A hell of a lot of nodes have a size. Anything with orange drag points in the 2D/3D tab is a thing that has a size, and that size should be visible in the inspector panel so that it can be accurately edited. It may not have a """`size`""" but it certainly has a """size""".

**Reply by AThousandShips** [reply]
In the cases where they have a `size` property you can just edit it, but in other cases that doesn't make any sense, you can't set the size of, for example, a `Polygon2D`, because while the extents can be calculated it doesn't have a size to *edit*

So while having some form of way to display the bounds of nodes, like the bounding box, having it be displayed as a property in the inspector is misleading

**Comment by timothyqiu** [comment]
> I am trying to create multiple collision shapes of the same width, but there's no way for me to see their width in the editor.

A `CollisionShape2D` node does not have size. It's the `shape` resource it uses that has a size. Click the button next to it expands the inspector, and you can now view & edit the size.

<img width="313" height="203" alt="image" src="https://github.com/user-attachments/assets/5e84fa17-8f3f-46b9-a086-05c07d59bc22" />
